usecases/seed_usecase.go

```diff
@@ -21,86 +25,84 @@ type SeedUseCase struct {
 }
 
 func (usecase *SeedUseCase) SeedMarbleAdmins(ctx context.Context, firstMarbleAdminEmail string) error {
-	return usecase.transactionFactory.Transaction(ctx, func(tx repositories.Executor) error {
-		_, err := usecase.userRepository.CreateUser(ctx, tx, models.CreateUser{
-			Email: firstMarbleAdminEmail,
-			Role:  models.MARBLE_ADMIN,
-		})
+	exec := usecase.executorFactory.NewExecutor()
+	logger := utils.LoggerFromContext(ctx)
 
-		// ignore user already added
-		if repositories.IsUniqueViolationError(err) {
-			return models.ErrIgnoreRollBackError
-		}
-		return err
+	_, err := usecase.userRepository.CreateUser(ctx, exec, models.CreateUser{
+		Email: firstMarbleAdminEmail,
+		Role:  models.MARBLE_ADMIN,
 	})
-}
 
-func (usecase *SeedUseCase) SeedZorgOrganization(ctx context.Context, zorgOrganizationId string) error {
-	exec := usecase.executorFactory.NewExecutor()
-	_, err := usecase.organizationCreator.CreateOrganizationWithId(
-		ctx,
-		zorgOrganizationId,
-		"Zorg",
-	)
-	if repositories.IsUniqueViolationError(err) {
-		err = nil
-	}
-
-	if err != nil {
+	// ignore user already added
+	if err != nil && !repositories.IsUniqueViolationError(err) {
 		return err
 	}
+	logger.InfoContext(ctx, fmt.Sprintf("Created marble admin user with email %s (or already exists)", firstMarbleAdminEmail))
+	return nil
+}
 
-	// assign test s3 bucket name to zorg organization
-	testBucketName := "marble-backend-export-scheduled-execution-test"
-	err = usecase.organizationRepository.UpdateOrganization(ctx, exec, models.UpdateOrganizationInput{
-		Id:                         zorgOrganizationId,
-		ExportScheduledExecutionS3: &testBucketName,
-	})
-	if err != nil {
-		return err
+// This method is supposed to be used as a script when starting the server, not from the API
+// Hence it does not enforce any authorization, since there is also no user credentials context
+func (usecase *SeedUseCase) CreateOrgAndUser(ctx context.Context, input models.InitOrgInput) error {
+	if input.OrgName == "" {
+		return errors.New("Cannot create organization or org admin with empty name in CreateOrgAndUser")
 	}
-
-	// add Admin user Jean-Baptiste Emanuel Zorg
-	_, err = usecase.userRepository.CreateUser(ctx, exec, models.CreateUser{
-		Email:          "jbe@zorg.com", // Jean-Baptiste Emanuel Zorg
-		Role:           models.ADMIN,
-		OrganizationId: zorgOrganizationId,
-		FirstName:      "Jean-Baptiste Emanuel",
-		LastName:       "Zorg",
-	})
-	if repositories.IsUniqueViolationError(err) {
-		err = nil
+	if input.AdminEmail != "" {
+		_, err := mail.ParseAddress(input.AdminEmail)
+		if err != nil {
+			return errors.New(fmt.Sprintf("Invalid email address %s in CreateOrgAndUser", input.AdminEmail))
+		}
 	}
+	logger := utils.LoggerFromContext(ctx)
+	exec := usecase.executorFactory.NewExecutor()
+
+	var targetOrg models.Organization
+	allOrgs, err := usecase.organizationRepository.AllOrganizations(ctx, exec)
```

Comment:
When reading this, I see "lazy to do a getByName repository method" 😛

Reply:
non mais blague à part, un peu lazy pour le coup c'est vrai mais en même temps, comme le nombre d'org devrait rester sous contrôle dans un premier temps, je trouve ça ok de réutiliser la méthode de répo existante et ajouter 4 lignes de loop ensuite.